tasks/push-rpm-manifests-to-pyxis/push-rpm-manifests-to-pyxis.yaml

```diff
@@ -0,0 +1,155 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: push-rpm-manifests-to-pyxis
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that extracts sboms from pull specs and pushes them to Pyxis.
+  params:
+    - name: pyxisJsonPath
+      description: Path to the JSON string of the saved Pyxis data in the data workspace
+      type: string
+    - name: pyxisSecret
+      type: string
+      description: |
+        The kubernetes secret to use to authenticate to Pyxis. It needs to contain two keys: key and cert
+    - name: server
+      type: string
+      description: The server type to use. Options are 'production' and 'stage'
+      default: production
+    - name: concurrentLimit
+      type: string
+      description: The maximum number of images to be processed at once
+      default: 4
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot spec json file resides
+  volumes:
+    - name: workdir
+      emptyDir: {}
+  steps:
+    - name: download-sbom-files
+      image:
+        quay.io/redhat-appstudio/release-service-utils:8bf56a04aaeb371f4a822d2b76520e9bdcacfb26
+      volumeMounts:
+        - mountPath: /workdir
+          name: workdir
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        PYXIS_FILE="$(workspaces.data.path)/$(params.pyxisJsonPath)"
+        if [ ! -f "${PYXIS_FILE}" ] ; then
+            echo "No valid pyxis file was provided."
+            exit 1
+        fi
+
+        IMAGEURLS=($(cat "${PYXIS_FILE}" | jq -r '.components[].pyxisImages[].containerImage' | sort | uniq))
+
+        mkdir /workdir/sboms
+        cd /workdir/sboms
+
+        for i in ${!IMAGEURLS[@]}; do
+          echo "Fetching sbom for image: ${IMAGEURLS[$i]}"
+          cosign download sbom --output-file "${IMAGEURLS[$i]}.json" "${IMAGEURLS[$i]}"
```

Comment:
Could you maybe at least do some basic substitution in the filename? I'm actually surprised if this works - I am unable to create a file with `/` in the filename on my system. `@` is also not great in a filename, but probably will work. So I would just do something like `FILE=$(echo $IMAGEURLS[$i] | tr "/@:" ---). And of course do the same in the second step.

Reply:
updated